pkg/models/spending.go

```diff
@@ -163,19 +161,19 @@ func (e *Spending) CalculateNextContribution(
 		// Otherwise we can simply look at how much we need vs how much we already have.
 		amountNeeded := myownsanity.Max(0, perSpendingAmount-currentAmount)
 		// And how many times we will have a funding event before our due date.
-		numberOfContributions := fundingSchedule.GetNumberOfContributionsBetween(now, nextRecurrence, timezone)
+		numberOfContributions := funding.GetNumberOfContributionsBetween(now, nextRecurrence, timezone)
 		// Then determine how much we would need at each of those funding events.
 		totalContributionAmount = amountNeeded / myownsanity.Max(1, numberOfContributions)
 	}
 
-	// Update the spending item with our calculated contribution amount.
-	e.NextContributionAmount = totalContributionAmount
-
 	// If the current nextRecurrence on the object is in the past, then bump it to our new next recurrence.
 	if e.NextRecurrence.Before(now) {
 		e.LastRecurrence = &e.NextRecurrence
 		e.NextRecurrence = nextRecurrence
 	}
 
-	return nil
+	// Update the funding item with our calculated contribution amount.
+	nextFunding := funding.GetNextFunding(now, timezone)
+	nextFunding.NextContributionAmount = totalContributionAmount
+	return &nextFunding, nil
```

Comment:
This wont work with multiple funding schedules that overlap. Maybe I should tinker with ripping out this entire bit of code and we just use the forecasting package? It would create a circular import but I could work around that.

But that way the calculations are exactly the same for forecasting and for the actual calculation. The only difference being the forecast assumes something is always spent on the day specified. Where as the real use case isn't concerned with spending (except when it comes to funding).

Reply:
Updated this so now it does work with multiple. Not using the forecasting code yet, it still needs work and should not be relied on just yet.